.github/workflows/build-common.yml

```diff
@@ -379,12 +371,10 @@ jobs:
         with:
           # only push cache for one matrix option per OS since github action cache space is limited
           cache-read-only: ${{ inputs.cache-read-only || matrix.smoke-test-suite != 'tomcat' }}
-          # gradle enterprise is used for the build cache
-          gradle-home-cache-excludes: caches/build-cache-1
 
       - name: Build
         # running suite "none" compiles everything needed by smoke tests without executing any tests
-        run: ./gradlew :smoke-tests:test -PsmokeTestSuite=none --no-daemon ${{ inputs.no-build-cache && ' --no-build-cache' || '' }}
+        run: ./gradlew :smoke-tests:test -PsmokeTestSuite=none ${{ inputs.no-build-cache && ' --no-build-cache' || '' }}
```

Comment:
`--no-daemon` was only used in two places, so removed for consistency

Reply:
I think the `--no-daemon` was used here to reduce memory usage.

Reply:
I didn't check the commit history, but I suspect the issue was that the tests sometimes got an OOM. So it was split in 2 parts where in the first part we compile the tests with `--no-daemon` (`-PsmokeTestSuite=none` doesn't run any tests) so whatever memory gradle used for building gets freed. Second step just runs the tests and hopefully has gradle consuming less memory. It is possible that the OOM isn't an issue any more. If that is the case we could remove the `Build` step from that workflow and just keep `Test`.

Reply:
oh! I missed the second step below, that makes sense, I'll add this back